core/domain/stats_jobs_one_off.py

```diff
@@ -1373,3 +1373,51 @@ def reduce(key, items):
             yield (key, len(items))
         else:
             yield (key, items)
+
+
+class ExplorationMissingStatsAudit(jobs.BaseMapReduceOneOffJobManager):
+    """A one-off job for finding explorations that are missing stats models."""
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(exp):
+        if exp.deleted:
+            return
+
+        exp_versions = list(python_utils.RANGE(1, exp.version + 1))
+        exp_stats_model_ids = (
+            stats_models.ExplorationStatsModel.get_entity_id(exp.id, version)
+            for version in exp_versions)
+        exp_stats_models = stats_models.ExplorationStatsModel.get_multi(
+            exp_stats_model_ids, include_deleted=True)
+
+        for exp_stats_model, exp_version in python_utils.ZIP(
+                exp_stats_models, exp_versions):
+            if exp_stats_model is None:
+                yield ('missing', (exp.id, exp_version))
+            elif exp_stats_model.deleted:
+                yield ('deleted', (exp.id, exp_version))
+            else:
+                yield ('present', (exp.id, exp_version))
```

Comment:
Maybe, just for safety, include here a check that the exp_version matches the version of the exp_stats_model.